eda-env/Lib/site-packages/parso/cache.py

```
@@ -0,0 +1,275 @@
+                except OSError:  # silently ignore all failures
+                    continue
+    else:
+        return True
+
+
+def _touch(path):
+    try:
+        os.utime(path, None)
+    except FileNotFoundError:
+        try:
+            file = open(path, 'a')
```

Ruff (0.11.9)

Use a context manager for opening files

(SIM115)